### PR TITLE
Feature/#155 course file view

### LINKED ISSUE #155: As a lecturer or as a student, i want to see all materials associated to a course in the course overview - grouped into materials uploaded by students and materials uploaded by lecturers
DEP: #154 

AC:
- [x] In the course overview, there is a box containing a scrollable list of all materials that were uploaded from lecturers, and another scrollable box, that contains a list of all materials from students
- [x] Clicking on the title of a material will "open" it directly (download / open in browser if link)
- [x] Authorship of the file is not provided by the list

_Hint: Use the [course overview mock](https://www.figma.com/file/6yo4FXp2lLtE4uZvI0lsFr/Paper-Prototype?node-id=135%3A8) if you want_

### PR DESCRIPTION
**Pull request by caustt**
# Resolves #155

This PR changes / adds / introduces the feature of ....

Acceptance Criteria:

- [x] In the course overview, there is a box containing a scrollable list of all materials that were uploaded from lecturers, and another scrollable box, that contains a list of all materials from students
- [x] Clicking on the title of a material will "open" it directly (download / open in browser if link)
- [x] Authorship of the file is not provided by the list

Testing steps:
- automated test cases

Steps needed for migration:
- None
---

**Student:**
image

**Lecturer:**
image

_Originally posted by @adrianjost in https://github.com/hpi-swt2/lecture-portal/issues/155#issuecomment-573415818_

Definition Of Done:
- [ ] [Class-Diagram](https://www.lucidchart.com/invitations/accept/705a122b-58a9-4dfa-8b16-c936bc0a46bc) updated

### CODE REVIEW

File: app/views/courses/_filelist.html.erb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<ul class="list-group mb-2">
+  <% files.each do |file| %>
+    <li class="list-group-item d-flex justify-content-between align-items-center">
+      <div>
+        <%= fa_icon 'file', :class => "text-info" %>
+        <%= link_to file.filename, uploaded_file_path(file), target: "_self", class:"btn btn-link" %>
+      </div>
+      <div>
+        <% if @current_user == file.author || @current_user == @course.creator %>
+          <button class="btn btn-sm btn-outline-danger">TODO ADD DELETE ACTION</button>
```
Comment (adrianjost) [inline app/views/courses/_filelist.html.erb]: The delete Action will be done in another PR I guess?

File: spec/views/courses/show.html.erb_spec.rb [hunk 2]
```diff
@@ -7,13 +7,16 @@
...
     sign_in @lecturer
   end
 
   it "renders title and description of the course " do
     render
-    expect(rendered).to have_text("SWT2")
-    expect(rendered).to have_text("ruby")
+    expect(rendered).to match(@course.name)
+    expect(rendered).to match(@course.description)
+    expect(rendered).to match(@course.status)
```
Comment (adrianjost) [inline spec/views/courses/show.html.erb_spec.rb]:
```suggestion
```

File: app/views/courses/_filelist.html.erb (hunk 1 above)
Comment (caustt) [reply app/views/courses/_filelist.html.erb]: yes, now you can find it [here](https://github.com/hpi-swt2/lecture-portal/pull/191).
